index.html

```diff
@@ -89,47 +151,288 @@
+          .attr("y", -42)
+          .attr("x", margin.top - (height/2))
+          .attr("dy", ".71em")
+          .style("text-anchor", "middle")
+          .text("Math proficiency");
+   
+
+
+        /*     
+          .append("text")
+          .attr("transform", "rotate(-90)")
+          .attr("y", 0 - margin.left)
+          .attr("dy", ".7em")
+          .style("text-anchor", "end")
+          .text("Math proficiency");
+*/
+
+    
     // EVENT HANDLERS
 
-    d3.select('#year-range').on('change', function () {
-      options.yearRange = d3.event.target.value;
+    d3.select('#categories').on('change', function () {
+      options.categoryRange = d3.event.target.value;
       update();
     });
   }
 
   function update() {
     // TRANSFORM
-    txData = data.slice();
+    schoolData = data.slice();
+/*
+    data.forEach(function (d) { 
+      d.filtered = d.Category;
+      var schoolData = data.filter(function (d) {return d.filtered;});
+    })
+*/
 
-    if (options.yearRange !== 'all') {
-      var yearRange = options.yearRange.split('-');
-      txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
+    if (options.categoryRange !== 'all') {
+     // var categoryRange = options.categoryRange.split('');
+      schoolData = schoolData.filter(function (d) {
+      //  return d.Category >= categoryRange[0] && d.Category <= categoryRange[1];
+          return d.Category === d3.event.target.value;
       });
+
     }
 
     // SCATTERPLOT
 
+   // svg.selectAll('.point')
+    //  .remove();
+
+  var points = svg.selectAll('.point') //point is used instead of circles, in case different shapes are used
+    .data(schoolData);
+
+    //console.log(points)
+
+    points.enter().append('circle')  //add points if points don't exist
+      .attr('class', 'point')
+      .attr('r', 5)
+
+    points    //all points
+      .attr('cx', function (d) { return x(d.Reading_proficient/100); })
+      .attr('cy', function (d) { return y(d.Math_proficient/100); });
+
+    points.exit().remove();
+
     // HISTOGRAMS
 
-    // REGRESSION LINE
+    //X-AXIS
+
+    var hist_x = d3.layout.histogram()
+          .value(function(d) { return d.Reading_proficient/100; })
+          .range([0, d3.max(schoolData, function(d){ return d.Reading_proficient/100; }) ])
+          .bins(x.ticks(10))
+
+    var layout = hist_x(schoolData);
+
+    for(var i = 0; i < layout.length; i++) {
+    var bin = layout[i];
+    var colorScale = d3.scale.category20();
+    svg.selectAll("circle")
+      .data(bin, function(d) { return d.Reading_proficient/100; })
+      .style("fill", function() {  return colorScale(i); })
+
+     }
+
+    var maxLength = d3.max(layout, function(d) { return d.length });
+    var heightScale = d3.scale.linear()
+      .domain([0, maxLength])
+      .range([0, height/5]);
+
+    var xScale = d3.scale.ordinal()
+      .domain(d3.range(10))
+      .rangeBands([0, width], 0.2)
+
```

Comment:
The thing is, we don't actually need a new x scale. We can use the existing one. This will address the issue you can see where the bars aren't aligned with the tick marks — the tick marks are set based on the existing x scale.

This will require changing how you calculate the `x` and the `width` attributes of the rects below. Instead of using the index (`i`) of each datum, we'll instead use the data given to us by the histogram layout (i.e. `x` and `dx`). Refer to this week's class demonstration for reference.